### ISSUE TITLE
Observability: Observability for DPF

### ISSUE THREAD

**Issue by ouphi** [agera-edc/DataSpaceConnector#173]
Story for feature #40
After #161, #172

upstream: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/936

## Description

The goal of this story is to validate that end to end observability for DPF works as expected. If it is not the case, it should be implemented.

## Acceptance Criteria
- [x] Valuable DPF traces & metrics should be present with proper context propagation
- [ ] Existing samples can be used to show end to end observability (distributed tracing) for data transfers where both the Control Plane and the Data Plane are involved
- [x] Integration tests for tracing to make sure that the expected spans are present.

## Tasks
- [x] Make sure that expected spans are present. If not, add them.
- [x] Make sure that DPF traces are tested with integration tests. If not, add them.
- [ ] Add instructions in sample to visualise traces.

**Comment by marcgs** [comment]
Waiting for following PRs to get merged:
- [Testing: Add integration tests for traces by ouphi · Pull Request #1035 · eclipse-dataspaceconnector/DataSpaceConnector (github.com)](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1035)

- [Feature/902/161 dpf system test by algattik · Pull Request #204 · agera-edc/DataSpaceConnector (github.com)](https://github.com/agera-edc/DataSpaceConnector/pull/204)

**Comment by marcgs** [comment]
Sample 4.3 is broken. A separate PR #227  will fix it. Proposing to add instructions for visualizing file transfer traces in that story: https://github.com/agera-edc/DataSpaceConnector/pull/227#issuecomment-1100665190.

**Comment by marcgs** [comment]
Upstream PR: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1162
